### ISSUE TITLE
version 4.2 coverage combine empties .coverage

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#511]
*Originally reported by* **Anonymous**

----------------------------------------

I've been using coverage through tox environments to combine coverage for various python/dependency versions at https://github.com/jcb91/IPython-notebook-extensions and https://github.com/ipython-contrib/jupyter_contrib_nbextensions. This was working fine, until the release of coverage 4.2, since which, `coverage combine` results in empty `.coverage` files. I'm unsure exactly why that's happening though.

`coverage debug sys` give the following output:

```
              version: 4.2
             coverage: /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python3.4/site-packages/coverage/__init__.py
           cover_dirs: /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python3.4/site-packages/coverage
           pylib_dirs: /opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4
               tracer: CTracer
 plugins.file_tracers: -none-
         config_files: .coveragerc
                       setup.cfg
         configs_read: setup.cfg
            data_path: /Users/josh/Documents/workspace/IPython-notebook-extensions/.coverage
               python: 3.4.4 (default, Mar  2 2016, 03:34:28) [GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)]
             platform: Darwin-14.5.0-x86_64-i386-64bit
       implementation: CPython
           executable: /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/bin/python3.4
                  cwd: /Users/josh/Documents/workspace/IPython-notebook-extensions
                 path: 
                       /Users/josh/Documents/workspace/IPython-notebook-extensions/tests
                       /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python34.zip
                       /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python3.4
                       /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python3.4/plat-darwin
                       /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python3.4/lib-dynload
                       /opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4
                       /opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/plat-darwin
                       /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python3.4/site-packages
                       /Users/josh/Documents/workspace/IPython-notebook-extensions/src
          environment: PYTHONHASHSEED = [174841001 (bb)](https://bitbucket.org/ned/coveragepy/commits/174841001)
                       PYTHONPATH = /Users/josh/Documents/workspace/IPython-notebook-extensions/tests
                       PYTHONUNBUFFERED = yes
         command_line: .tox/report/bin/coverage report --debug=config,dataio,dataop,plugin,sys
         source_match: -none-
    source_pkgs_match: -none-
        include_match: -none-
           omit_match: -none-
          cover_match: /Users/josh/Documents/workspace/IPython-notebook-extensions/.tox/report/lib/python3.4/site-packages/coverage
          pylib_match: /opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4
```

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/511

**Comment by nedbat** [comment]
Did you read the backwards-incompatible change in 4.2?  https://bitbucket.org/ned/coveragepy/annotate/fee2fd97520c0bea5a5e58e54cd7d0b23dcc0dd2/CHANGES.rst?at=default&fileviewer=file-view-default#CHANGES.rst-44

I'm not sure that's what's going on here, but if it is, then adding `--append` to your combine step will fix it.

**Comment by nedbat** [comment]
I believe this is fixed in [34ae57c51cfb (bb)](https://bitbucket.org/ned/coveragepy/commits/34ae57c51cfb), it would be great if you could test it out for me.

**Comment by nedbat** [comment]
This fix was released as part of Coverage.py 4.3.
